pkg/namestore/namestore.go

```diff
@@ -48,8 +48,8 @@ type nameStore struct {
 }
 
 func (x *nameStore) Acquire(name, id string) error {
-	if err := identifiers.Validate(name); err != nil {
-		return fmt.Errorf("invalid name %q: %w", name, err)
+	if err := validator.Validate(name); err != nil {
```

Comment:
```suggestion
	if err := identifiers.ValidateDocker(name); err != nil {
```

Reply:
The difference from containerd id validator has to be explained in godoc

Reply:
should we have a docker compat option in nerdctl config file in that way customer can configure that and the commands need not change.
Here i can have a check if the flag is set.

Reply:
i can add the flag as part of this pr.

Reply:
> The difference from containerd id validator has to be explained in godoc

will add an explanation in godoc.

Reply:
> should we have a docker compat option in nerdctl config file in that way customer can configure that and the commands need not change.

Sounds overengineering

Reply:
ok then can just change the names and add a doc.